examples/sumcheck.rs

```diff
@@ -0,0 +1,137 @@
+//! Example: multilinear sumcheck proofs.
+//!
+//! Sumcheck proofs allow to prove that, given a multilinear polynomial $g$
+//! defined over field $\mathbb{F}$, the sum of $g$ over boolean hypercube
+//! is equal to $C$.
+//!
+//! In the interactive version of the protocol, the number of rounds is equal
+//! to the number of variables. In each round, the prover sums the
+//! "round polynomial" over all the variables except for the first variable,
+//! and gets a linear polynomial described by two points. Then, it sends the
+//! description to the verifier, gets a challenge from the verifier, fixes the
+//! first variable of round polynomial on that challenge and obtains the next
+//! round polynomial (with one fewer variable).
+//!
+//! The verifier in each round checks whether the sum of received polynomial
+//! over boolean hypercube is equal to the expected value set in the previous
+//! round. If not, it rejects. In the final round, it checks whether the
+//! evaluation of the received polynomial at the challenge is equal to
+//! evaluation of the $g$ at the tuple of all challenges. If not, it rejects.
+//!
+//! In the Fiat-Shamir-transformed sumcheck, we do the same, with a minor
+//! optimization to reduce the transcript size. The prover, sends only one
```

Comment:
```
//! In the Fiat-Shamir-transformed sumcheck, we do the same, with a minor
//! optimization to reduce the transcript size.
```

Why do you say it concerns the Fiat-Shamir transformation? This can be done at the IOP level. I'd remove this sentence.